File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>s;
        for(auto num: nums)
        {
            s.insert(num);
        }
        int count = 0;
        for(auto num: nums)
        {
            if(s.find(num - 1) == s.end() )
            {
                int n = num ;
                int temp = 1;
                while(s.find(n+1) != s.end())
                {
                    n += 1;
                    temp += 1;
                }
                count = max(temp, count);
            }
        }
        return count;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for n in s:
            if n - 1 not in s:
                x = n 
                count = 0
                while x in s:
                    count += 1
                    x += 1
                res = max(count, res)
        return res
